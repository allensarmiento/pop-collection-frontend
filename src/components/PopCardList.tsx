import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import { Pop } from '../state/types';
import PopCard from './PopCard';

const useGridStyles = makeStyles(({ breakpoints }) => {
  return {
    root: {
      [breakpoints.up('xs')]: {
        justifyContent: 'center',
      },
    },
  };
});

type CardListProps = {
  pops: Array<Pop>;
};

const CardList = ({ pops }: CardListProps) => {
  const gridStyles = useGridStyles();

  return (
    <Grid classes={gridStyles} container spacing={4} wrap={'wrap'}>
      {!pops.length ? <h3>There are no pops!</h3> : (
        pops.map((pop, index) => (
          <Grid key={index} item>
            <PopCard
              title={pop.name}
              subtitle={pop.category}
              image={pop.image}
              color={pop.color}
            />
          </Grid>
        ))
      )}
    </Grid>
  );
};

export default CardList;
