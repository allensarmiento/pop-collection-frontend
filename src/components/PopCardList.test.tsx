import { shallow } from 'enzyme';
import PopCardList from './PopCardList';

const filteredPops = [{
  name: 'Name 1',
  color: '#fff',
  category: 'Category 1',
  image: '',
}];

it('renders without crashing', () => {
  expect(shallow(<PopCardList pops={filteredPops} />)).toMatchSnapshot();
});
