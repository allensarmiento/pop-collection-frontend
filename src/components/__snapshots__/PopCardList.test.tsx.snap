// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<WithStyles(ForwardRef(Grid))
  classes={
    Object {
      "root": "makeStyles-root-1",
    }
  }
  container={true}
  spacing={4}
  wrap="wrap"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="0"
  >
    <PopCard
      color="#fff"
      image=""
      subtitle="Category 1"
      title="Name 1"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
